platform/core/src/services/HangingProtocolService/HangingProtocolService.ts

```diff
@@ -175,7 +175,21 @@ class HangingProtocolService {
   public getProtocolById(id: string): HangingProtocol.Protocol {
     const protocol = this.protocols.get(id);
 
-    return protocol;
+    if (protocol instanceof Function) {
+      try {
+        const { protocol: generatedProtocol } = this._getProtocolFromGenerator(
+          protocol
+        );
+
+        return generatedProtocol;
+      } catch (error) {
+        console.warn(
```

Comment:
Is ```console.error better here?

Reply:
Is ```console.error``` better here?

Reply:
I think the reasoning behind using a warn rather than an error, is because we are handling this and not totally failing the execution of the app. @sedghi might have an opinion on this